fw/access_log.c

```diff
@@ -0,0 +1,430 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2022 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+#include "access_log.h"
+#include "connection.h"
+#include "server.h"
+#include "http.h"
+#include "lib/common.h"
+
+/* This thing describes access log format.
+ * - FIXED => fixed string, passed as is
+ * - UNTRUNCATABLE => this expression will never be truncated
+ * - TRUNCATABLE => this expression will be truncated if it does not fit
+ *   log buffer
+ *
+ * !!!
+ * If you add new UNTRUNCATABLE/TRUNCATABLE field don't forget to also
+ * set appropriate variable/array value in do_access_log_req()
+ !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ ! Adding TRUNCATABLE field will compile !
+ !  but won't work without changing the  !
+ !       code that fills its value       !
+ !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ */
+#define ACCESS_LOG_LINE(FIXED, UNTRUNCATABLE, TRUNCATABLE) \
+	FIXED("[tempesta fw] ")                            \
```

Comment:
We already have `[tempesta fw]` defined as `__BNR` in `log.h`.

BTW maybe we should move the contents of access_log.h to log.h to keep all logging declarations in the same place.